tests/pytests/test_json.py

```diff
@@ -1152,3 +1152,179 @@ def test_mod5608(env):
         env.expect('FT.CREATE', 'idx', 'ON', 'HASH', 'PREFIX', 1, 'd', 'SCHEMA', 'id', 'TAG', 'num', 'NUMERIC').equal('OK')
         waitForIndex(env, 'idx')
         _, cursor = env.cmd('FT.AGGREGATE', 'idx', "*", 'LOAD', 1, 'num', 'WITHCURSOR', 'MAXIDLE', 1, 'COUNT', 300)
+
+def testTagAutoescaping(env):
+
+    env = Env(moduleArgs = 'DEFAULT_DIALECT 5')
+
+    conn = getConnectionByEnv(env)
+    # We are using ',' as tag SEPARATOR to get the same results of HASH index
+    env.cmd('FT.CREATE', 'idx', 'ON', 'JSON', 
+            'SCHEMA', '$.tag', 'AS', 'tag', 'TAG', 'SEPARATOR', ',')
+
+    # create sample data
+    conn.execute_command('JSON.SET', 'doc:1', '$', r'{"tag": "abc:1"}')
+    conn.execute_command('JSON.SET', 'doc:2', '$', r'{"tag": "xyz:2"}')
+    conn.execute_command('JSON.SET', 'doc:4', '$', r'{"tag": "abc:1-xyz:2"}')
+    conn.execute_command('JSON.SET', 'doc:5', '$', r'{"tag": "joe@mail.com"}')
+    conn.execute_command('JSON.SET', 'doc:6', '$', r'{"tag": "tag with {brackets}"}')
+    conn.execute_command('JSON.SET', 'doc:7', '$', r'{"tag": "abc:1|xyz:2"}')
+    conn.execute_command('JSON.SET', 'doc:8', '$', r'{"tag": "_12@"}')
+    conn.execute_command('JSON.SET', 'doc:9', '$', r'{"tag": "-99999"}')
+    conn.execute_command('JSON.SET', 'doc:10', '$', r'{"tag": "ab(12)"}')
+    conn.execute_command('JSON.SET', 'doc:11', '$', r'{"tag": "a|b-c d"}')
+    conn.execute_command('JSON.SET', 'doc:12', '$', r'{"tag": "_@12\\345"}')
+    conn.execute_command('JSON.SET', 'doc:13', '$', r'{"tag": "$literal"}')
```

Comment:
Add a case with a literal star (to cover all `tag_invalid_punct`)

Reply:
OK. Test added: 
- [test_json.py:1199](https://github.com/RediSearch/RediSearch/pull/4433/commits/535c73ad18ddf43a9e092a8d280492eab4be4124#diff-cf0a34c0325b41a586cc82d8cf8a3b5b4d1c0750c42805a155fe57fc48b34ca5R1199)
-  [test_json.py:1244](https://github.com/RediSearch/RediSearch/pull/4433/commits/535c73ad18ddf43a9e092a8d280492eab4be4124#diff-cf0a34c0325b41a586cc82d8cf8a3b5b4d1c0750c42805a155fe57fc48b34ca5R1244)
- [test_json.py:1252](https://github.com/RediSearch/RediSearch/pull/4433/commits/535c73ad18ddf43a9e092a8d280492eab4be4124#diff-cf0a34c0325b41a586cc82d8cf8a3b5b4d1c0750c42805a155fe57fc48b34ca5R1252)
- [test_tags.py:427](https://github.com/RediSearch/RediSearch/pull/4433/commits/535c73ad18ddf43a9e092a8d280492eab4be4124#diff-b71b7402df7c4b83c0e5690ef9c4d133ddb694ca43d2371f648bcfe5f23267a2R427)
- [test_tags.py:530](https://github.com/RediSearch/RediSearch/pull/4433/commits/535c73ad18ddf43a9e092a8d280492eab4be4124#diff-b71b7402df7c4b83c0e5690ef9c4d133ddb694ca43d2371f648bcfe5f23267a2R530)
- [test_tags.py:547](https://github.com/RediSearch/RediSearch/pull/4433/commits/535c73ad18ddf43a9e092a8d280492eab4be4124#diff-b71b7402df7c4b83c0e5690ef9c4d133ddb694ca43d2371f648bcfe5f23267a2R547)
- [test_tags.py:569](https://github.com/RediSearch/RediSearch/pull/4433/commits/535c73ad18ddf43a9e092a8d280492eab4be4124#diff-b71b7402df7c4b83c0e5690ef9c4d133ddb694ca43d2371f648bcfe5f23267a2R569)
- [test_aggregate_params.py:111](https://github.com/RediSearch/RediSearch/pull/4433/commits/535c73ad18ddf43a9e092a8d280492eab4be4124#diff-4dfcf4e06e9ba073365dc052e7e19bc26a7f6347e3310e636d7cf718d8ae5360R111)